client/src/components/Auth/AuthForm.tsx

```diff
@@ -1,121 +1,163 @@
+    hasError: emailHasError,
+    valueChangeHandler: emailChangeHandler,
+    valueBlurHandler: emailBlurHandler,
+    reset: emailResetHandler,
+  }: IUserInput = useInput((value: string) => value.includes("@"));
+
+  const {
+    value: password,
+    isValid: passwordIsValid,
+    hasError: passwordHasError,
+    valueChangeHandler: passwordChangeHandler,
+    valueBlurHandler: passwordBlurHandler,
+    reset: passwordResetHandler,
+  }: IUserInput = useInput((value: string) => value.trim() !== "");
 
   const [isLogin, setIsLogin] = useState(true);
-  const [isLoading, setIsLoading] = useState(false);
 
   const switchAuthModeHandler = () => {
     setIsLogin((prevState) => !prevState);
   };
 
+  let formIsValid = false;
+  if (isLogin) {
+    if (emailIsValid && passwordIsValid) formIsValid = true;
+  } else {
+    if (nameIsValid && emailIsValid && passwordIsValid) formIsValid = true;
+  }
+
   const submitHandler = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
 
-    const email = emailInputRef.current?.value;
-    const password = passwordInputRef.current?.value;
-    const name = nameInputRef.current?.value;
-    console.log(email,name,password);
-    // optional: Add validation
+    if (!formIsValid) {
+      return;
+    }
 
-    setIsLoading(true);
-    let url;
+    let url: string;
     if (isLogin) {
-      url =
-        'http://localhost:9000/user/login';
+      url = loginURL;
     } else {
-      url =
-        'http://localhost:9000/user/register';
+      url = registerURL;
     }
-    try 
-    {
-        const response = await Axios({
-            method: 'POST',
-            url: url,
-            data: {
-                name,
-                email,
-                password
-            }
-        });
 
-        if (response.status === 200)
-        {
-            // Could also use history.push('/login')
-            setIsLoading(false);
-            dispatch(authActions.login(response.data))
-            navigate('/');
-        }
-        else
-        {   
-            setIsLoading(false);
-            console.log("unspecified");
-        }
-    } 
-    catch (error) 
-    {
-        setIsLoading(false);
-        const errObj = error as AxiosError
-        const err = errObj.response?.data;
-        const response = err as AxiosError;
-        const notify = () => toast.error(response.message);
-        notify();
+    signIn({
+      method: "POST",
+      url: url,
+      data: {
+        name,
+        email,
+        password,
+      },
+    })
+
+    nameResetHandler();
+    emailResetHandler();
+    passwordResetHandler();
+  };
 
+  useEffect(() => {
+    if(errorMessage?.length === 0) {
+      if(data.user.name.trim() !== ''){
+        dispatch(authActions.login(data));
+        dispatch(
+            toastActions.showToast({
+              type: Title.SUCCESS,
+              message: "Login Successful",
+            })
+          );
+      }
+      navigate('/');
```

Comment:
use navigate.push so that the history is maintained

Reply:
I googled this and found that push was required with useHistory..navigate() by default works the same as history.push().